File: Examples/CSharp/ModifyingAndConvertingImages/ExpandOrCropAnImage.cs
using Aspose.Imaging.ImageOptions;
using System;

/*
This project uses the Automatic Package Restore feature of NuGet to resolve the Aspose.Imaging for .NET API 
reference when the project is built. Please check https://learn.microsoft.com/en-us/nuget/resources/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from https://releases.aspose.com/, 
install it, and then add its reference to this project. For any issues, questions, or suggestions, 
please feel free to contact us using https://forum.aspose.com/
*/

namespace Aspose.Imaging.Examples.CSharp.ModifyingAndConvertingImages
{
    class ExpandOrCropAnImage
    {
        public static void Run()
        {
            Console.WriteLine("Running example ExpandOrCropAnImage");
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_ModifyingAndConvertingImages();

            // Load an image into an Image instance and cache its data.
            using (RasterImage rasterImage = (RasterImage)Image.Load(dataDir + "aspose-logo.jpg"))
            {
                rasterImage.CacheData();
                // Create a Rectangle that defines the X, Y, width, and height of the region, then save the output image.
                Rectangle destRect = new Rectangle { X = -200, Y = -200, Width = 300, Height = 300 };
                rasterImage.Save(dataDir + "Grayscaling_out.jpg", new JpegOptions(), destRect);
            }

            Console.WriteLine("Finished example ExpandOrCropAnImage");
        }
    }
}

File: Examples/VisualBasic/ModifyingAndConvertingImages/ExpandOrCropAnImage.vb
Imports Aspose.Imaging.ImageOptions

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Imaging for .NET API reference 
'when the project is build. Please check https://Docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Imaging for .NET API from http://www.aspose.com/downloads, 
'Install it and then add its reference to this project. For any issues, questions or suggestions 
'Please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace Aspose.Imaging.Examples.VisualBasic.ModifyingAndConvertingImages
    Class ExpandOrCropAnImage
        Public Shared Sub Run()
            ' ExStart:ExpandOrCropAnImage
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_ModifyingAndConvertingImages()

            ' Load an image in an instance of Image and  Setting for image data to be cashed
            Using rasterImage As RasterImage = DirectCast(Image.Load(dataDir & "aspose-logo.jpg"), RasterImage)
                rasterImage.CacheData()
                ' Create an instance of Rectangle class and define X,Y and Width, height of the rectangle and Save output image 
                Dim destRect As New Rectangle() With {.X = -200, .Y = -200, .Width = 300, .Height = 300}
                rasterImage.Save(dataDir & "Grayscaling_out.jpg", New JpegOptions(), destRect)
            End Using
            ' ExEnd:ExpandOrCropAnImage
        End Sub
    End Class
End Namespace